### PR TITLE
graphman: new command to hastly delete a subgraph

### PR DESCRIPTION
**Pull request by tilacog**
Introduces a new graphman command to promptly delete all data from a deployment, even if it is running.

TODO: 
- [x] Decide the name and placement for this command
- [x] Documentation

### CODE REVIEW

File: node/src/bin/manager.rs [hunk 1]
```diff
@@ -237,6 +237,29 @@ pub enum Command {
...
+        /// List only current version
+        #[clap(long, short)]
+        current: bool,
+        /// List only pending versions
+        #[clap(long, short)]
+        pending: bool,
+        /// List only used (current and pending) versions
+        #[clap(long, short)]
+        used: bool,
+        /// Skips confirmation prompt
```
Comment (lutter) [inline node/src/bin/manager.rs:259]: Should be `Skip` to stay in the imperative style of the rest of the help text.

File: node/src/bin/manager.rs [hunk 2]
```diff
@@ -1097,6 +1120,31 @@ async fn main() -> anyhow::Result<()> {
...
+            pending,
+            used,
+            force,
+        } => {
+            let sender = ctx.notification_sender();
+            let (store, mut pools) = ctx.store_and_pools();
+            let subgraph_store = store.subgraph_store();
+            let primary_pool = pools
+                .remove(&*PRIMARY_SHARD)
+                .expect("there is a primary pool");
```
Comment (lutter) [inline node/src/bin/manager.rs:1135]: There's a `ctx.store_and_primary()` which will do most of this for you

File: node/src/manager/commands/drop.rs [hunk 3]
```diff
@@ -0,0 +1,74 @@
+use crate::manager::{
+    deployment::{Deployment, DeploymentSearch},
+    display::List,
+    prompt::prompt_for_confirmation,
+};
+use graph::anyhow::{self, bail};
+use graph_store_postgres::{connection_pool::ConnectionPool, NotificationSender, SubgraphStore};
+use std::sync::Arc;
+
+/// Finds, unassings, record and remove matching deployments.
```
Comment (lutter) [inline node/src/manager/commands/drop.rs:10]: type: 'unassigns'

File: node/src/manager/commands/info.rs [hunk 4]
```diff
@@ -5,7 +5,7 @@ use graph_store_postgres::{connection_pool::ConnectionPool, Store};
 
 use crate::manager::deployment::{Deployment, DeploymentSearch};
 
-fn find(
+pub fn find(
```
Comment (lutter) [inline node/src/manager/commands/info.rs:8]: Might be better to make this a method on `DeploymentSearch` if it's used in more than one place

**Review by lutter — APPROVED** [review]
Nice! I really like the user interaction. Just address the few comments I left, and then this is good to go.

As for the command name: after looking at the commands, I agree it's much better to make this a new command than to add it to `graphman remove`.

File: node/src/bin/manager.rs (hunk 1 above)
Comment (tilacog) [reply node/src/bin/manager.rs]: Fixed in 7ff6aa8ba

File: node/src/bin/manager.rs (hunk 2 above)
Comment (tilacog) [reply node/src/bin/manager.rs]: Fixed in 9ccee3a82

File: node/src/manager/commands/info.rs (hunk 4 above)
Comment (tilacog) [reply node/src/manager/commands/info.rs]: Fixed in b2adf2922

**Pull request merged**